### DISCUSSION TITLE
pocketbase + next behind nginx

### DISCUSSION THREAD

**Discussion by Sand-NGR** [pocketbase/pocketbase#4000] [Q&A] [answered]
I'm using a VPS. i've set nginx as reverse proxy. in the docs there is a nginx config and it says to run this:
```
/root/pb/pocketbase serve yourdomain.com
```

when I run this it tells me `2023/12/20 14:38:24 listen tcp 0.0.0.0:443: bind: address already in use`. which is true. It is taken by nginx which is listening on port 443:

```
// sudo lsof -i :443 results in:

COMMAND  PID  USER   FD   TYPE DEVICE SIZE/OFF NODE NAME
nginx   9643  root    6u  IPv4  73132      0t0  TCP *:https (LISTEN)
nginx   9644 nginx    6u  IPv4  73132      0t0  TCP *:https (LISTEN)
nginx   9645 nginx    6u  IPv4  73132      0t0  TCP *:https (LISTEN)
```
how am i supposed to this? any sample code for nginx config with pocketbase and next.js deployment anywhere?

**Comment by ganigeorgiev** [comment, marked as answer]
`./pocketbase serve yourdomain.com` works only when you follow the [Minimal setup guide](https://pocketbase.io/docs/going-to-production/#minimal-setup), aka. when you don't use nginx.

If you use nginx, 80 and 443 ports are usually taken by the proxy and you need to start the PocketBase process on some other local port:

```sh
# the default http address is 127.0.0.1:8090
./pocketbase serve

# if you want to specify a different local port use the --http flag
./pocketbase serve --http='12.0.0.1:3000'
```
Note that when using reverse proxy it is up to you to manage the certificate. This is usually done  via [certbot](https://certbot.eff.org/).

**Reply by Sand-NGR** [reply]
understood. I have managed the SSL with certbot but there is a problem. In the docs there is a nginx config which is good but the `proxy_pass` in it is set to `http://127.0.0.1:8090`.  this is my nginx config:

```
server {
    server_name mydomain.dev;

    location / {
        proxy_pass http://127.0.0.1:8090;

        proxy_set_header Host $host;
        proxy_set_header Connection '';
        proxy_read_timeout 360s;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
  }


    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/mydomain.dev/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/mydomain.dev/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot

}

server {
    if ($host = mydomain.dev) {
        return 301 https://$host$request_uri;
    } # managed by Certbot


    listen 80;
    server_name mydomain.dev;
    return 404; # managed by Certbot


}

```

if we proxy to `http://127.0.0.1:8090` how can I see the admin panel? should mydomain.dev/_/ show me the admin panel?

**Reply by Sand-NGR** [reply]
never mind it was a stupid problem with the CDN God damn it. sorry for taking your time.
